src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/service/NonAssociationsService.kt

```diff
@@ -57,6 +57,9 @@ class NonAssociationsService(
   }
 
   fun createNonAssociation(createNonAssociationRequest: CreateNonAssociationRequest): NonAssociationDTO {
+    val createdBy = authenticationHolder.authenticationOrNull?.userName
```

Comment:
if you don't get these exceptions actually being thrown in live, then you could replace by `authenticationHolder.username`.  If you are handling `UserInContextMissingException` then might be better for the `HmppsAuthenticationHolder` to throw a single exception instead of the two it does at present (or rather the two will be subclasses of a single exception) and then you could just let the holder throw its own one

Reply:
your tests would then be a lot simpler as you could just mock the `username` call on the holder too